sqoop/sqoop.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+#    Copyright 2018 Google, Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# This script installs Apache Sqoop (http://sqoop.apache.org/) on a Google Cloud
+# Dataproc cluster. Hive-Hcatalog is not installed automatically and if you want
+# to run hive jobs than please use this script with hive-hcatalog.sh init action.
+# HBase libraries are not installed automatically so in order to run imports to 
+# BigTable using sqoop, please run this init action with bigtable.sh.
+
+set -euxo pipefail
+
+readonly SQOOP_HOME='/usr/lib/sqoop'
+readonly SQOOP_CODE_LINK='https://github.com/apache/sqoop'
+readonly BIGTABLE_HBASE_CLIENT='bigtable-hbase-1.x-hadoop-1.3.0.jar'
+readonly HBASE_BIGTABLE_DL_LINK="http://central.maven.org/maven2/com/google/cloud/bigtable/bigtable-hbase-1.x-hadoop/1.3.0/${BIGTABLE_HBASE_CLIENT}"
+
+function update_apt_get() {
+  for ((i = 0; i < 10; i++)); do
+    if apt-get update; then
+      return 0
+    fi
+    sleep 5
+  done
+  return 1
+}
+
+function err() {
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $@" >&2
+  return 1
+}
+
+function install_sqoop() {
+  git clone ${SQOOP_CODE_LINK} ${SQOOP_HOME} && cd ${SQOOP_HOME} && ant \
+    || err 'Compiling Sqoop from source code failed.'
+  echo "export PATH=\"${SQOOP_HOME}/bin:$PATH\"" >> /etc/profile
```

Comment:
Source builds are pretty much forbidden in our init actions for performance reasons.

If Sqoop is not available as a binary distribution I would drop this PR and resubmit it when it's fixed in our Bigtop repo.

Reply:
@pmkc if you would like as temporary solution I can pack compiled sqoop in some lightweight docker container ? WDYT?

Reply:
@szewi I think it will be a little while until this is in Bigtop. For now, I suggest we host a mirror on GCS.

I'd prefer not to build from trunk or any non-release commit, since non-release commits are not guaranteed to be stable. Please run `git cherry-pick e13dd21209c26316d43350a23f5d533321b61352` onto `release-1.4.7-rc0`, resolve the merge conflicts, run Sqoop's tests, and build a jar. I can then put that jar in the appropriate bucket.

If that's more complicated than just minor merge conflicts, then we can build directly from `e13dd21209c26316d43350a23f5d533321b61352` and host that.

Either way, this init action should be hermetic and should not build from trunk.

Reply:
@karth295 I cherry-picked `e13dd21209c26316d43350a23f5d533321b61352` onto release-1.4.7-rc0 resolved conflicts, but tests were not passing. I investigated this and decided to first cherry-pick commit `18445290810b1df035e06fb074064d6b9c1d6e90` which was solving the issues with tests and also allowed to apply then `e13dd21209c26316d43350a23f5d533321b61352` without conflicts. Then I compiled sqoop on a clean dataproc cluster and uploaded sqoop to publicly available bucket. Most probably you would like to move sqoop.jar to your own place on google storage, so please let me know about final location, then I will update URL properly. 

Sqoop is being delivered with `bin/` directory containing required executable scripts, so right now sqoop code is downloaded to /usr/lib/sqoop, but no compilation is done. I changed the source code location to my sqoop fork with the branch `sqoop-bigtable-client` containing changes I have applied, so the end user will have exact sources used for creating sqoop jar in /usr/lib/sqoop. Let me know if you are fine with this.